scripts/automation/lecterntrade.arucas

```diff
@@ -0,0 +1,488 @@
+            player.setSelectedSlot(i);
+            return true;
+        }
+    }
+    if (slot != null) {
+        if (0 <= slot - 36 && slot - 36 <= 8) {
+            player.setSelectedSlot(slot - 36);
+            return true;
+        } else {
+            preslot = player.getSwappableHotbarSlot();
+            player.swapPlayerSlotWithHotbar(slot);
+            player.setSelectedSlot(preslot);
+            return true;
+        }
+    }
+    throwRuntimeError("No pickaxe item was found");
+}
+
+fun swapToItem(itemString) {
+    if (itemString == "air") {
+        return false;
+    }
+    foreach (i : [ 0, 1, 2, 3, 4, 5, 6, 7, 8 ]){
+        if (player.getItemForPlayerSlot(i).getId() == itemString || player.getItemForPlayerSlot(i).getId().contains(itemString) ){
+            player.setSelectedSlot(i);
+            return false;
+        }
+    }
+    slot = player.getSlotFor(ItemStack.of(Material.of(itemString)));
+    retVal = false;
+    if (slot == null) {
+        throwRuntimeError("No " + itemString + " was found");
+    }
+    if (player.getItemForSlot(slot).getCount() >= 1) {
+        retVal = true;
+    }
+    if (slot != null) {
+        if (0 <= slot - 36 && slot - 36 <= 8) {
+            player.setSelectedSlot(slot - 36);
+            return retVal;
+        } else {
+            preslot = player.getSwappableHotbarSlot();
+            player.swapPlayerSlotWithHotbar(slot);
+            player.setSelectedSlot(preslot);
+            return retVal ;
+        }
+    }
+
+}
+fun getRandomOffset(){
+    randrange = 10;
+    center = 0.4;
+    return (random(randrange * 2) - randrange) / randrange * 0.8 + 0.1 - 0.5;
+}
+fun placeBlock(){
+    while (true) {
+        if (world.getBlockAt(bx,by,bz).getId()=='lectern'){
+            return true;
+        }
+        swapToItem('lectern');
+        //player.lookAtPos(bx, by, bz);
+        player.interactBlock(bx + 0.5 + getRandomOffset(),by, bz+0.5 + getRandomOffset(),"up",bx,by-1,bz, false);
+        sleep(100);
+    }
+}
+isPaused = true;
+fun togglePause(){
+    isPaused = !isPaused;
+    if(isPaused){
+        player.message('Search is now paused');
+    }
+    else{
+        player.message('Search is now active');
+    }
+
+}
+enchantlists = {};
+
+fun addToEnchantsParsed(enchantsPreSeparate){
+    sep = '-';
+    split = enchantsPreSeparate.split('-');
+    if(Math.mod(len(split),3) == 0){
+        for(ii = 0; ii< len(split)/3;ii++){
+            enchantName = split.get(3*ii);
+            lvl = split.get(3*ii+1).toNumber();
+            price = split.get(3*ii+2).toNumber();
+            player.message('Search list added: '+ enchantName + ' '+ lvl + ' ' +price);
+            enchantlists.put(enchantName, {'lvl': lvl, 'price': price});
+        }
+    }
+    else {
+        player.message('Given enchantments is not multiplier of 3');
+        player.message('ex) /tradeSetup add fortune-1-13-protection-2-24');
+    }
+}
+trial = 0;
+isWanted = false;
+isExecuting = false;

```

Comment:
The fact that your globals are scattered all over the place is sure difficult to read